pkgs/development/libraries/fltk/generic.nix

```diff
@@ -0,0 +1,115 @@
+, lib
+, fetchFromGitHub
+, cmake
+, pkg-config
+, doxygen
+, texlive
+, xorgproto
+, xlibsWrapper
+, libXi
+, freeglut
+, expat
+, libGL
+, libGLU
+, libjpeg
+, zlib
+, libpng
+, libXinerama
+, libXft
+, libXcursor
+, cairo
+, pcre
+, libpthreadstubs
+, libXdmcp
+, pango
+, fribidi
+, freetype
+, libtiff
+, AGL
+, Cocoa
+, GLUT
+
+  # Passed by versions
+, version
+, rev
+, sha256
+, extraPatches ? [ ]
+, buildOptions
+, ...
+}:
+let
+  inherit (lib) optional optionals optionalString versionAtLeast;
+  onOff = (value: if value then "ON" else "OFF");
+  withOption = (name: lib.attrsets.attrByPath [ name ] false buildOptions);
+  optionToCmakeOption = (value: if builtins.typeOf value == "string" then value else onOff value);
+in
+stdenv.mkDerivation {
+  pname = "fltk";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "fltk";
+    repo = "fltk";
+    inherit rev sha256;
+  };
+
+  patches = optionals stdenv.hostPlatform.isDarwin [ ./nsosv.patch ]
+    ++ extraPatches;
+
+  nativeBuildInputs = [ cmake pkg-config ]
+    ++ optional (withOption "BUILD_HTML_DOCUMENTATION") doxygen
+    ++ optional (withOption "BUILD_PDF_DOCUMENTATION") texlive;
+
+  buildInputs = [ expat ]
+    ++ optionals (withOption "USE_GL") [ libGL libGLU ]
+    ++ optional (withOption "USE_SYSTEM_LIBJPEG") libjpeg
+    ++ optional (withOption "USE_SYSTEM_ZLIB") zlib
+    ++ optional (withOption "USE_SYSTEM_LIBPNG") libpng
+    ++ optional (withOption "USE_XINERAMA") libXinerama
+    ++ optional (withOption "USE_XFT") libXft
+    ++ optional (withOption "USE_XCURSOR") libXcursor
+    ++ optionals (withOption "CAIRO" || withOption "CAIROEXT") [ cairo pcre libpthreadstubs libXdmcp ]
+    ++ optionals (withOption "USE_PANGO") [ pango fribidi pcre libpthreadstubs libXdmcp ]
+    # TODO properly move these out by cmake flags
+    ++ optionals stdenv.hostPlatform.isDarwin [ AGL Cocoa GLUT ];
+
+  # TODO verify which of these are actually needed?
+  propagatedBuildInputs = [
+    stdenv.cc # fltk-config script references build-time compiler
+    xorgproto
+  ]
+  ++ (
+    if stdenv.hostPlatform.isDarwin then [
+      freetype
+      libtiff
+    ] else [
+      xlibsWrapper
+      libXi
+      freeglut
+    ]
+  );
+
+  cmakeFlags = lib.attrsets.mapAttrsToList (name: value: "-DOPTION_${name}=${optionToCmakeOption value}") buildOptions;
+
+  postBuild = optionalString (withOption "BUILD_HTML_DOCUMENTATION" || withOption "BUILD_PDF_DOCUMENTATION") ''
+    make docs
+  '';
+
+  postInstall = optionalString (withOption "BUILD_EXAMPLES") ''
+    for example in bin/examples/*; do
+      binname="$(basename "$example")"
+      if [ ! "$binname" == "$(basename "$binname" .cxx)" ]; then
+        continue;
+      fi
+      cp "$example" $out/bin/"$binname"
+    done
+  '';
+
+  meta = with lib; {
+    description = "A C++ cross-platform lightweight GUI library";
+    homepage = "https://www.fltk.org";
+    platforms = with platforms; linux ++ darwin;
+    license = licenses.lgpl2;
```

Comment:
lgpl2Only or lpgpl2Plus?

Reply:
Clarified it for now, not sure if it makes sense to give them their own license instead? See https://www.fltk.org/COPYING.php.

Reply:
Maybe add a comment to see their site for exceptions.

Reply:
I already did that, the view here is just outdated.

```nix
# LGPL2 with static linking exception
# https://www.fltk.org/COPYING.php
license = licenses.lgpl2Only;
 ```